src/parser.rs

```diff
@@ -1213,6 +1224,15 @@ impl Parser {
         )
     }
 
+    pub fn loop_statement(&mut self) -> NodeId {
+        let span_start = self.position();
+        self.keyword(b"loop");
+        let block = self.block(BlockContext::Curlies);
+        let span_end = self.get_span_end(block);
```

Comment:
Looking good but a quesiton we should discuss with @sophiajt is how we want to consistently track the span ends. 
We have: 
- peeking the inner parse like here via `self.get_span_end()`
- manual computation (based on known lengths or byte offsets)
- just querying `self.position()` after the parse

Reply:
Yeah, I found this slightly awkward as well. Maybe `self.get_span(span_start)` could be added, which would internally use `self.position()` and create a span using the given span start?